jupyterhub_moss/spawner.py

```diff
@@ -233,29 +234,30 @@ async def _get_slurm_info_resources(self):
                 errmsg = "Missing required resource counter in Slurm partition: {}"
                 raise KeyError(errmsg.format(partition))
 
-        return (resources_display, resources_info)
+        partitions_info = {}
+        for partition, description in self.partitions.items():
+            partitions_info[partition] = resources_info[partition]
+            # use data from Slurm as base and overwrite with manual configuration settings
+            partitions_info[partition].update(description)
+
+        return (resources_display, partitions_info)
 
     @staticmethod
     async def create_options_form(spawner):
         """Create a form for the user to choose the configuration for the SLURM job"""
-        resources_display, resources_info = await spawner._get_slurm_info_resources()
+        resources_display, partitions_info = await spawner._get_partitions_info()
 
-        # Combine all partition info as a dict
-        partition_info = {}
-        default_partition = None
-        for partition in spawner.partitions:
-            # use data from Slurm as base and overwrite with manual configuration settings
-            partition_info[partition] = resources_info[partition]
-            partition_info[partition].update(spawner.partitions[partition])
-            spawner.partitions[partition] = partition_info[partition]
-
-            if partition_info[partition]["simple"] and default_partition is None:
-                default_partition = partition
```

Comment:
Combination of `self.partitions` and `sinfo` moved in `_get_partitions_info`.
Only the retrieval of the `default_partition` is kept here.